### PR TITLE
[#66 #94] backend API & AJAX frontend

### LINKED ISSUE #65: Answering a question should be done via AJAX
[Priority: Low, Risk: Low]

### LINKED ISSUE #66: Convert select features to use AJAX to minimize page reloads
[Priority: Low, Risk: Low]

To minimize page reloads and improve user experience, we should convert certain functions to use ajax requests.

### LINKED ISSUE #67: Make comments submit via AJAX
[Priority: Low, Risk: Low]

### LINKED ISSUE #94: As a developper, I want to access the data via an API
[Priority: Medium, Risk:Low]

For AJAX to work, with or without view, we need to implement all our data output in a RESTful API format, feeding JSON via endpoints allowing for quick, seamless and dynamic UI.

Acceptance Test : 
- [x] I can access comments data via the /api/comments endpoints
- [x] I can access questions data via the /api/questions endpoints
- [x] I can access answers data via the /api/answers endpoints
- [x] I can access votes data via the /api/{type}/{id}/votes endpoint (included in #64)

### PR DESCRIPTION
**Pull request by WindOfRussia**
Pulled from branch, reviewed copious amounts of changes, ran unit tests and tried to break it on my own.
tldr : good code, tests pass, here's the pull request

### here's a summary *(you should copy it into the merge commit description)*

Implements a tiny API with our data on the backend
that is consumed by the new (mostly) async frontend made in vue.js

Contributors: @Evi1NoOb @Incommensurable @outofphasor @PeterGhimself @bnduwi 
Reviewers: @WindOfRussia

works on #64 
closes #65 
closes #67 
closes #66 
closes #94

### CODE REVIEW

**Comment by NicoChampagne** [comment]
Congrats on the hard work guys
